### ISSUE TITLE
#iterations with accum_grad > 1

### ISSUE THREAD

**Issue by kan-bayashi** [espnet/espnet#777]
Hi, @ShigekiKarita.
In chainer’s trainer, the number of interation count is every updated when `update_core` called.
So, if we use `accum_grad = 2`, the actual iteration will be halved.
I have modified this issue as follows:
https://github.com/espnet/espnet/blob/75f7a14751d0bb51e96faa42fa313a595ef2b1a2/espnet/tts/pytorch_backend/tts.py#L150-L153

**Comment by ShigekiKarita** [comment]
thanks! @Fhrozen can you also fix this in #774 ?

**Comment by Fhrozen** [comment]
Ok, i will add this modification, thanks for reporting it

**Comment by Fhrozen** [comment]
Fixed in #774

**Comment by kehanlu** [comment]
Hi,
does it mean that if we increase the number of gradient accumulation, the training time will be longer since one iteration means one `optimizer.step()` but not one `loss.backward()`?  Then each data point will be seen `gradient_acc*epoch` times.

I saw other implementation(e.g. speechbrain) that one iteration means one `loss.backward()` and after `gradient_acc` times of backpropagation it performs `optimizer.step()`, so each data point will only be seen `epoch` times.

I just want to make sure the implementation difference for fair comparison. Thanks in advance!
